### ISSUE TITLE
Scanning different places

### ISSUE THREAD

**Issue by seagate1992** [AHAAAAAAA/PokemonGo-Map#2608]
We are 4 friends playing pokemon go, and for the moment I run the command 4 time on 4 different port because we wan't scan 4 different places.

But is there an other way to do this without opening 4 ports and run 4 time the command line ?

Thanks :-)

**Comment by loulou1310** [comment]
Yes,

You can use a database and only open 1 port.
Here is the link to the tutorial to install the DB.
https://github.com/AHAAAAAAA/PokemonGo-Map/wiki/How-to-start-using-a-MySQL-Server

And here the config.ini file that I use for this purpose.
```
db-type: mysql
db-host: 127.0.0.1
db-name: PKM
db-user: root
db-pass: root

scan-delay:1
num-threads:1

host:0.0.0.0
port:8793
```

And the launch command :
`start "LOC1" "C:\Python27\python.exe" runserver.py -u api_ptc_acc_1 -p api_ptc_pwd -l "LAT,LNG" -st 7 -P 8793`
`timeout 3`
`start "LOC2" "C:\Python27\python.exe" runserver.py -u api_ptc_acc_2 -p api_ptc_pwd -l "LAT,LNG" -st 7 -P 8793`
`timeout 3`
`start "LOC3" "C:\Python27\python.exe" runserver.py -u api_ptc_acc_3 -p api_ptc_pwd -l "LAT,LNG" -st 7 -P 8793`
`timeout 3`
`start "LOC4" "C:\Python27\python.exe" runserver.py -u api_ptc_acc_3 -p api_ptc_pwd -l "LAT,LNG" -st 7 -P 8793`

**Comment by colwellkr** [comment]
Do you really need three different accounts to be able to do this?

**Comment by dudenell** [comment]
Yes you need 3 different accounts to do this. Your limited on your rate and each one will be limited a certain amount.

**Comment by mahavidhan** [comment]
I am not using Maria but generally I use a command that starts with `runserver.py` and the whole lot of parameters next to it. Wondering what is `start` is it a command? and what to enter in "LOC1" when there is already a LAT,LNG in the parameters. Also what is timout 3. Is this a batch script or something?

**Comment by mahavidhan** [comment]
I just completed the steps given in the tutorial and things are working fine. I have setup my config.ini file without location an giving the command `python runserver.py -st 2 -l "mylocation"` I just need some clarity on the launch commands you posted. 

1. Do I have to open four cmd windows for the four different commands?
2. What is timeout 3
3. Why does your command start with `start "LOC1"`. What location to give here if I am giving the LAT/LNG anyways  in the command line?
3. where is the database stored?

**Comment by talentedmrripley** [comment]
If you run the commands from a batch with as listed with start in front of them it will spawn the additional windows.  You could also run all but one of them with the -ns flag which just runs the search without running the server process.

**Comment by mahavidhan** [comment]
Thank you for your reply but i am really not understanding because of confusion.
What I understood is if I copy-paste those four commands in a notepad file and save it as .bat extension and run it then it will open 4 different windows. Is that what you are trying to say?
What do you mean by running search without running the server process?

**Comment by talentedmrripley** [comment]
@mahavidhan Yes, if you run the commands @loulou1310 described from a single batch file it will launch all four processes.  

You can further refine it if you want.  By default each time you launch a process it launches the back end search and the front end display.  Since your goal is to only have one open port there's no reason to run four front end maps.  So you have two options.  You can launch all four processes with the -ns flag which is only the search and then launch one process with -os which is only the front end.  Or you can launch 3 processes with the -ns flag and one without.  Both accomplish basically the same task.  Keep in mind if you use -os method or one without -ns which ever coordinates you specify on the map is where the map will center every time its loaded.  

Since you're sharing it with multiple people I'd be tempted to run 4 -ns and one -os because that also should prevent someone from turning on geolocation and moving one of the search areas inadvertently.  

Lastly for each process that will do searching you will likely need a separate login.  It appears that Niantic has started throttling requests per account to 3 per second.  For each search worker I suggest you do -t 3 and -sd 1.1 (that's three threads with a delay between each call of 1.1 seconds).  You can either specify that per command line or uncomment and fill in the appropriate sections in the config.ini.

**Comment by seagate1992** [comment]
Yes it work.

I tried with 2 accounts. Here is the  first command
`python runserver.py -a Google -u MyUser -p MyPass -l "Location1" -st 7`

`python runserver.py -a Google -u MyUser2 -p MyPass2 -l "Location2" -st 7 -ns`

When i'm on Location1, and change for a third Location (Location3), the scan on Location2 continue and everything is ok.
After that, if i'm on Location2 and change for a fourth Location (Location4), the scan on Location3 stop and return on Location1

here is my config.ini

`db-type: mysql
db-host: 127.0.0.1
db-name: pkmndb
db-user: user
db-pass: pwd

scan-delay:1
num-threads:1

host:w.x.y.z
port:8793

gmaps-key: myKey
`

**Comment by mahavidhan** [comment]
@talentedmrripley thanks for the info. all confusion cleared. I am ready to search the pokemons. Two more questions:
1. Will it work if I create a database on my website and in the config.ini file if I pass the required info? So that when my friends visit my website say mywebsite.com:5000 they will be able to view the map? If yes, do I need to keep my cmd window open with the commands executed?
2. I just downloaded an app from play store (https://play.google.com/store/apps/details?id=com.jv.pokemapgo) that does exactly what this map does. Is there an android app release on github? 
Thanks for your help.

**Comment by talentedmrripley** [comment]
@mahavidhan Yes that should work fine.  You don't need to create a DB, it will use SQLite by default.  You can setup MySQL/MariaDB if you wish though.  And yes, you'll need to leave the command window running, assuming its a Windows server.  If you're using a Linux box you could launch it with something like screen so that you don't h ave to leave the window open.

As far as the Android thing, I'm not sure.

**Comment by colwellkr** [comment]
There is an android port of this map available for you to build here: https://github.com/omkarmoghe/Pokemap
